aea/test_tools/test_skill.py

```diff
@@ -0,0 +1,264 @@
+        dialogue_reference: Optional[  # pylint: disable=unused-argument
+            Tuple[str, str]
+        ] = None,
+        message_id: Optional[int] = None,  # pylint: disable=unused-argument
+        target: Optional[int] = None,  # pylint: disable=unused-argument
+        to: Optional[Address] = None,
+        sender: Address = "counterparty",
+        **kwargs,
+    ) -> Message:
+        """
+        Quickly create an incoming message with the provided attributes.
+
+        For any attribute not provided, the corresponding default value in message is used.
+
+        :param message_type: the type of the message
+        :param dialogue_reference: the dialogue_reference
+        :param message_id: the message_id
+        :param target: the target
+        :param performative: the performative
+        :param to: the 'to' address
+        :param sender: the 'sender' address
+        :param kwargs: other attributes
+
+        :return: the created incoming message
+        """
+        message_attributes = dict()
+        message_attributes["dialogue_reference"] = Dialogues._generate_dialogue_nonce()  # pylint: disable=protected-access
+        for arg_name in inspect.getfullargspec(self.build_incoming_message)[0][1:]:
+            if (
+                arg_name not in {"message_type", "to", "sender"}
+                and locals()[arg_name] is not None
+            ):
+                message_attributes[arg_name] = locals()[arg_name]
+        message_attributes.update(kwargs)
+
+        incoming_message = message_type(**message_attributes)
+        incoming_message.sender = sender
+        incoming_message.to = (
+            self.skill.skill_context.agent_address if to is None else to
+        )
+
+        return incoming_message
+
+    @staticmethod
+    def _complete_compact_message(
+        message: Tuple[Any, ...], last_incoming: Optional[bool], message_id: int
+    ) -> Tuple[Any, ...]:
+        new_message = message
+        if type(message[0]) != bool:  # pylint: disable=unidiomatic-typecheck
+            new_message = (
+                (False, *message) if last_incoming is True else (True, *message)
+            )
+        if type(new_message[1]) != int:  # pylint: disable=unidiomatic-typecheck
+            new_message_list = list(new_message)
+            new_message_list.insert(1, message_id - 1)
+            new_message = tuple(new_message_list)
+        return new_message
+
+    def build_dialogue(
+        self,
+        dialogues: Dialogues,
+        messages: Tuple[Tuple[Any, ...]],
+        counterparty: Address = "counterparty",
+    ) -> Dialogue:
+        """
+        Quickly create a dialogue.
+
+        The 'messages' argument is a tuple of "compact messages".
+        Each compact message is represented as a tuple (inc, target, performative, contents), where
+         - 'inc' is boolean, representing whether the message is incoming (True), or outgoing (False).
```

Comment:
I don't think creating these "compact messages" is a good idea. It is yet another concept which has to be understood and maintained. The developer might as well just provide the message object at this point...

Reply:
There is no concept. It is just a list of performatives and contents, which is the bare minimum of providing messages for a dialogue (with the option to specify targets and incoming/outgoing if the messages do not have the "standard" pattern of each targeting the one before and alternating between incoming/outgoing).

Reply:
I suggest we introduce it in `aea.protocols.dialogue.base` since it is effectively the representation of a message that is useful only in a dialogue.

Reply:
I disagree, different representations of the message should be close together. But let's see when it is applied throughout...